acpi/src/madt.rs

```diff
@@ -48,7 +50,55 @@ unsafe impl AcpiTable for Madt {
     }
 }
 
+pub type PhysToVirtFn = fn(u64) -> u64;
+
 impl Madt {
+    pub fn wakeup_aps(
+        &self,
+        apic_id: u32,
+        wakeup_vector: u64,
+        phys_to_virt: PhysToVirtFn,
```

Comment:
This crate does not assume every user will have a trivial physical to virtual translation mapping in place - this should use the user's `AcpiHandler` to create a mapping for the mailbox and unmap it once done with it.